### PR TITLE
CCM-5998: Extend linter to entire project

### PR DESCRIPTION
**Pull request by lapenna-bjss**
## Summary

This PR is to extend the linter setup to the entire project, following the initial implementation in the sandbox folder.

## Important changes

- Added lint to pre-commit hook
- A ticket will be created to investigate the "no-eval" errors

Additionally, I ran the ZAP pipeline to double-check that these changes didn't cause any issues.
image

## Reviews Required
* [x] Dev
* [ ] Test
* [ ] Tech Author
* [ ] Product Owner

## Checklist
* [x] Brief description of work completed, and any technical decisions made as part of the PR
* [x] PR link added as a comment to the relevant JIRA ticket
* [ ] PR link shared on Slack and/or Teams
* [ ] 2 reviews received
* [ ] Tester approval

### CODE REVIEW

File: proxies/shared/resources/jsc/500DuplicateTemplates.ExtractDuplicates.js [hunk 1]
```diff
@@ -1,3 +1,3 @@
 const body = context.getVariable("response.content");
-const message = JSON.parse(body).message;
+const {message} = JSON.parse(body);
```
Comment (simonlabarere) [inline proxies/shared/resources/jsc/500DuplicateTemplates.ExtractDuplicates.js:2]: Does this syntax actually work in Apigee? Having a green pipeline should be a good proof that it's working but I'm not sure why it won't start.

File: zap/scripts/authentication/get_bearer_token.js [hunk 2]
```diff
@@ -52,19 +52,19 @@ function authenticate(helper, paramsValues, credentials) {
     signer.update([headers, token].join(".").getBytes(StandardCharsets.UTF_8));
 
     const signature = encoder.encodeToString(signer.sign());
-    const jwt = headers + "." + token + "." + signature;
+    const jwt = `${headers}.${token}.${signature}`;
```
Comment (simonlabarere) [inline zap/scripts/authentication/get_bearer_token.js:55]: It's probably worth doing a zap scan against this branch to make it's okay with these changes.

Comment (lapenna-bjss) [reply zap/scripts/authentication/get_bearer_token.js:55]: I did run a zap scan, I added a screenshot in the description. Link to the scan: https://dev.azure.com/NHSD-APIM/API%20Platform/_build/results?buildId=235972&view=results

Comment (simonlabarere) [reply zap/scripts/authentication/get_bearer_token.js:55]: Ah yeah, sorry about that :D

File: proxies/shared/resources/jsc/500DuplicateTemplates.ExtractDuplicates.js (hunk 1 above)
Comment (simonlabarere) [reply proxies/shared/resources/jsc/500DuplicateTemplates.ExtractDuplicates.js]: This syntax does work on apigee and it's making the pipeline fail as expected.

**Comment by simonlabarere** [comment]
No longer required, we don't want to apply the linting rules to the apigee config. Other PR to be raised to run the linter as part of the pre-commit hook.

**Pull request closed without merging**